lib/Dialect/ESI/ESITranslations.cpp

```diff
@@ -130,26 +127,25 @@ LogicalResult ExportCosimSchema::emit() {
 
   // We need a sorted list to ensure determinism.
   llvm::sort(types.begin(), types.end(),
-             [](capnp::TypeSchema &a, capnp::TypeSchema &b) {
-               return a.capnpTypeID() > b.capnpTypeID();
-             });
+             [](auto &a, auto &b) { return a->typeID() > b->typeID(); });
 
   // Compute and emit the capnp file id.
   uint64_t fileHash = 2544816649379317016; // Some random number.
-  for (capnp::TypeSchema &schema : types)
-    fileHash =
-        llvm::hashing::detail::hash_16_bytes(fileHash, schema.capnpTypeID());
+  for (auto &schema : types)
```

Comment:
Nit: plz don't use auto in where it's not blatantly obvious what the type is and it's not an unreasonably complex type.